### PR TITLE
Feature/hydra 976 compression options

### PR DESCRIPTION
**Pull request by russorat**
Adding compression options to hdfs sinks.

### CODE REVIEW

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/S3AvroBatchSink.java [hunk 1]
```diff
@@ -111,6 +117,19 @@ public S3AvroOutputFormatProvider(S3AvroSinkConfig config, BatchSinkContext cont
       conf = Maps.newHashMap();
       conf.put(JobContext.OUTPUT_KEY_CLASS, AvroKey.class.getName());
       conf.put("avro.schema.output.key", config.schema);
+      if (config.compressionCodec != null && !config.compressionCodec.equals("None")) {
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/S3AvroBatchSink.java:120]: Can we ignore case when we check for 'None'.

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/S3AvroBatchSink.java [hunk 2]
```diff
@@ -111,6 +117,19 @@ public S3AvroOutputFormatProvider(S3AvroSinkConfig config, BatchSinkContext cont
...
+        conf.put("mapred.output.compress", "true");
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/S3AvroBatchSink.java:121]: Can we put "mapred.output.compress" and such in a static variable at the top of the class.

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/SnapshotFileBatchParquetSink.java [hunk 3]
```diff
@@ -77,6 +77,22 @@ protected void addFileProperties(FileSetProperties.Builder propertiesBuilder) {
       throw new RuntimeException("Error: Schema is not valid ", e);
     }
 
+    if (config.compressionCodec != null && !config.compressionCodec.equals("None")) {
+      switch (config.compressionCodec) {
+        case "Snappy":
+          propertiesBuilder.setOutputProperty("parquet.compression", "SNAPPY");
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/SnapshotFileBatchParquetSink.java:83]: Same here!!

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/SnapshotFileBatchParquetSink.java [hunk 4]
```diff
@@ -77,6 +77,22 @@ protected void addFileProperties(FileSetProperties.Builder propertiesBuilder) {
...
+          break;
+        case "GZip":
+          propertiesBuilder.setOutputProperty("parquet.compression", "GZIP");
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/SnapshotFileBatchParquetSink.java:86]: Same here

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/SnapshotFileBatchParquetSink.java [hunk 5]
```diff
@@ -77,6 +77,22 @@ protected void addFileProperties(FileSetProperties.Builder propertiesBuilder) {
...
+          break;
+        case "LZO":
+          propertiesBuilder.setOutputProperty("parquet.compression", "LZO");
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/SnapshotFileBatchParquetSink.java:89]: Same here

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TimePartitionedFileSetDataSetORCSink.java [hunk 6]
```diff
@@ -60,6 +62,31 @@ public void initialize(BatchRuntimeContext context) throws Exception {
   @Override
   protected void addFileSetProperties(FileSetProperties.Builder properties) {
     FileSetUtil.configureORCFileSet(config.schema, properties);
+    if (config.compressionCodec != null && !config.compressionCodec.equals("None")) {
+      LOG.info("Config: {}", config);
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TimePartitionedFileSetDataSetORCSink.java:66]: remove INFO log.

File: core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLBatchTestBase.java [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2015 Cask Data, Inc.
+ * Copyright © 2016 Cask Data, Inc.
```
Comment (nitinmotgi) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLBatchTestBase.java:2]: 2015, 2016

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TimePartitionedFileSetSink.java [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2015-2016 Cask Data, Inc.
+ * Copyright © 2016 Cask Data, Inc.
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TimePartitionedFileSetSink.java:2]: 2015, 2016

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TimePartitionedFileSetDatasetAvroSink.java [hunk 9]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2015-2016 Cask Data, Inc.
+ * Copyright © 2016 Cask Data, Inc.
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TimePartitionedFileSetDatasetAvroSink.java:2]: 2015, 2016

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/SnapshotFileBatchAvroSink.java [hunk 10]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2015 Cask Data, Inc.
+ * Copyright © 2016 Cask Data, Inc.
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/SnapshotFileBatchAvroSink.java:2]: 2015, 2016

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/S3AvroBatchSink.java [hunk 11]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2015 Cask Data, Inc.
+ * Copyright © 2016 Cask Data, Inc.
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/S3AvroBatchSink.java:2]: 2015, 2016

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TimePartitionedFileSetDatasetParquetSink.java [hunk 12]
```diff
@@ -56,6 +56,21 @@ public TimePartitionedFileSetDatasetParquetSink(TPFSParquetSinkConfig config) {
   @Override
   protected void addFileSetProperties(FileSetProperties.Builder properties) {
     FileSetUtil.configureParquetFileSet(config.schema, properties);
+    if (config.compressionCodec != null && !config.compressionCodec.equals("None")) {
+      switch (config.compressionCodec) {
+        case "Snappy":
```
Comment (nitinmotgi) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TimePartitionedFileSetDatasetParquetSink.java:61]: This code is being repeated, could we centralize this.

File: core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLTPFSTestRun.java [hunk 13]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2015-2016 Cask Data, Inc.
+ * Copyright © 2016 Cask Data, Inc.
```
Comment (nitinmotgi) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLTPFSTestRun.java:2]: 2015, 2016

File: core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLBatchTestBase.java [hunk 14]
```diff
@@ -141,6 +141,28 @@ public static void setupTest() throws Exception {
                       SSHAction.class);
   }
 
+  protected boolean checkFileType(TimePartitionedFileSet fileSet, String expectedType) throws IOException {
```
Comment (nitinmotgi) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLBatchTestBase.java:144]: This is not generic and wouldn't be able to handle the time partition specification.

File: core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/S3BatchSinkTest.java [hunk 15]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2015 Cask Data, Inc.
+ * Copyright © 2016 Cask Data, Inc.
```
Comment (nitinmotgi) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/S3BatchSinkTest.java:2]: 2015, 2016

File: hdfs-plugins/src/main/java/co/cask/hydrator/plugin/HDFSSink.java [hunk 16]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2015-2016 Cask Data, Inc.
+ * Copyright © 2016 Cask Data, Inc.
```
Comment (nitinmotgi) [inline hdfs-plugins/src/main/java/co/cask/hydrator/plugin/HDFSSink.java:2]: 2015, 2016

File: hdfs-plugins/src/test/java/co/cask/hydrator/plugin/HDFSSinkTest.java [hunk 17]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2015 Cask Data, Inc.
+ * Copyright © 2016 Cask Data, Inc.
```
Comment (nitinmotgi) [inline hdfs-plugins/src/test/java/co/cask/hydrator/plugin/HDFSSinkTest.java:2]: 2015, 2016

**Review by nitinmotgi — CHANGES_REQUESTED** [review]
Few minor comments. Please address them.

**Comment by russorat** [comment]
closing in favor of new PR

**Pull request closed without merging**